prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java

```diff
@@ -693,9 +693,17 @@ public APIResult registerExtensionMetadata(
     public APIResult getExtensionDefinition(
             @PathParam("extension-name") String extensionName) {
         checkIfExtensionServiceIsEnabled();
+        checkIfExtensionExists(extensionName);
         try {
-            return new APIResult(APIResult.Status.SUCCEEDED, ExtensionStore.get().getResource(extensionName,
-                    extensionName.toLowerCase() + EXTENSION_PROPERTY_JSON_SUFFIX));
+            ExtensionType extensionType = getExtensionType(extensionName);
+            if (ExtensionType.TRUSTED.equals(extensionType)) {
+                return new APIResult(APIResult.Status.SUCCEEDED, ExtensionStore.get().getResource(extensionName,
+                        extensionName.toLowerCase() + EXTENSION_PROPERTY_JSON_SUFFIX));
+            } else {
+                return new APIResult(APIResult.Status.SUCCEEDED,
+                        ExtensionStore.getMetaStore().getDetail(extensionName).getDescription());
```

Comment:
This is returning the short description which the user supplied at the time of registration. What should be returned is the content of file/files under META.